File: Math/CPP/largest_perimeter_triangle.cpp
#include <algorithm>
#include <vector>
using namespace std;

// 976. Largest Perimeter of a Triangle

class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        int ans = 0;
        sort(nums.begin(), nums.end());
        for (int i = nums.size() - 1; i - 2 >= 0; i--) {
            int a = nums[i], b = nums[i-1], c = nums[i-2];
            if (check(a,b,c)) ans = max(ans, a+b+c);
        }
        return ans;
    }

    bool check(int a, int b, int c) {
        return ((a + b) > c) && ((a + c) > b) && ((b + c) > a);
    }
};

File: Math/largest_perimeter_triangle.py
class Solution:
    def largestPerimeter(self, nums: list[int]) -> int:
        ans = 0
        nums.sort()
        for i in range(len(nums)-1, 1, -1):
            a, b, c = nums[i], nums[i-1], nums[i-2]
            if ((a + b) > c) and ((a + c) > b) and ((b + c) > a):
                ans = max(ans, a+b+c)
        return ans